elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/scaladsl/ElasticsearchSource.scala

```diff
@@ -9,14 +9,32 @@ import akka.stream.alpakka.elasticsearch._
 import akka.stream.scaladsl.Source
 import org.elasticsearch.client.RestClient
 import spray.json._
-import DefaultJsonProtocol._
 
+/**
+ * Scala API to create Elasticsearch sources.
+ */
 object ElasticsearchSource {
 
   /**
-   * Scala API: creates a [[ElasticsearchSourceStage]] that consumes as JsObject
+   * Creates a [[akka.stream.scaladsl.Source]] from Elasticsearch that streams [[OutgoingMessage]]s
+   * of Spray's [[spray.json.JsObject]].
+   * Alias of [[create]].
+   */
+  def apply(indexName: String,
+            typeName: String,
+            query: String,
+            settings: ElasticsearchSourceSettings = ElasticsearchSourceSettings())(
+      implicit client: RestClient
+  ): Source[OutgoingMessage[JsObject], NotUsed] = create(indexName, typeName, query, settings)
+
+  /**
+   * Creates a [[akka.stream.scaladsl.Source]] from Elasticsearch that streams [[OutgoingMessage]]s
+   * of Spray's [[spray.json.JsObject]].
    */
-  def apply(indexName: String, typeName: String, query: String, settings: ElasticsearchSourceSettings)(
+  def create(indexName: String,
+             typeName: String,
+             query: String,
+             settings: ElasticsearchSourceSettings = ElasticsearchSourceSettings())(
```

Comment:
Not sure what we have said in Alpakka in general, but default parameter is in general a real headache when it comes to maintaining binary compatibility, so that could be the reason we didn't have it here.

Reply:
For the future I mean, right now we don't guarantee bincomp here.

Reply:
Ok, I'll try to keep that in mind.